### PR TITLE
Support heco bsc

### PR DESCRIPTION
**Pull request by peterQin0826**
## What does this implement/fix? Explain your changes.
支持BSC/HECO

## Does this close any currently open issues?  
If it fixes a bug or resolves a feature request, be sure to link to that issue.
OneKeyHQ/TaskHub#975

## Pull request type
 
_Put an `x` in the boxes that apply_
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## Where has this been tested?
…

## Any other comments?
…

### CODE REVIEW

File: android/app/src/main/java/org/haobtc/onekey/adapter/HdWalletAssetAdapter.kt [hunk 1]
```diff
@@ -37,6 +38,22 @@ class HdWalletAssetAdapter(context: Context, data: List<BalanceCoinInfo?>?) : Ba
...
+          .setText(R.id.text_balance, getFormatBalance(item, 6))
+      }
+      item.coin.equals(Vm.CoinType.HECO.callFlag, true) -> {
+        helper.setImageDrawable(
+          R.id.imageView,
+          ResourcesCompat.getDrawable(helper.itemView.resources, R.drawable.vector_token_heco, null)
+        )
+        helper.setText(R.id.text_wallet_name, item.coin)
+          .setText(R.id.text_balance, getFormatBalance(item, 6))
+      }
```
Comment (QuincySx) [inline android/app/src/main/java/org/haobtc/onekey/adapter/HdWalletAssetAdapter.kt]: 使用 CoinDisplayUtils

File: android/app/src/main/java/org/haobtc/onekey/onekeys/homepage/WalletFragment.java [hunk 2]
```diff
@@ -506,8 +506,14 @@ private void toNext(int id) {
...
-                            .equalsIgnoreCase(Vm.CoinType.ETH.callFlag)) {
+                                    .callFlag
+                                    .equalsIgnoreCase(Vm.CoinType.ETH.callFlag)
+                            || value.getCoinType()
+                                    .callFlag
+                                    .equalsIgnoreCase(Vm.CoinType.BSC.callFlag)
+                            || value.getCoinType()
+                                    .callFlag
+                                    .equalsIgnoreCase(Vm.CoinType.HECO.callFlag)) {
                         SendEthActivity.start(getActivity(), value.getId());
```
Comment (QuincySx) [inline android/app/src/main/java/org/haobtc/onekey/onekeys/homepage/WalletFragment.java]: 使用 chaintype 判断

File: android/app/src/main/java/org/haobtc/onekey/onekeys/homepage/WalletFragment.java [hunk 3]
```diff
@@ -506,8 +506,14 @@ private void toNext(int id) {
...
-                            .callFlag
-                            .equalsIgnoreCase(Vm.CoinType.ETH.callFlag)) {
+                                    .callFlag
+                                    .equalsIgnoreCase(Vm.CoinType.ETH.callFlag)
+                            || value.getCoinType()
+                                    .callFlag
+                                    .equalsIgnoreCase(Vm.CoinType.BSC.callFlag)
+                            || value.getCoinType()
+                                    .callFlag
+                                    .equalsIgnoreCase(Vm.CoinType.HECO.callFlag)) {
```
Comment (QuincySx) [inline android/app/src/main/java/org/haobtc/onekey/onekeys/homepage/WalletFragment.java]: 使用 chaintype 判断

File: android/app/src/main/java/org/haobtc/onekey/onekeys/homepage/process/HdWalletDetailActivity.java [hunk 4]
```diff
@@ -171,6 +171,16 @@ private void inits() {
...
+            case BSC:
+                mImageTokenLogo.setImageDrawable(
+                        ResourcesCompat.getDrawable(
+                                getResources(), R.drawable.vector_token_bsc, null));
+                break;
+            case HECO:
+                mImageTokenLogo.setImageDrawable(
+                        ResourcesCompat.getDrawable(
+                                getResources(), R.drawable.vector_token_heco, null));
+                break;
```
Comment (QuincySx) [inline android/app/src/main/java/org/haobtc/onekey/onekeys/homepage/process/HdWalletDetailActivity.java:183]: 使用 AssetsLogo 类

File: android/app/src/main/java/org/haobtc/onekey/adapter/WalletListTypeAdapter.java [hunk 5]
```diff
@@ -48,28 +51,43 @@ protected void convert(BaseViewHolder helper, WalletInfo item) {
...
+                                    R.drawable.vector_token_heco,
                                     null));
                 }
+
+                Drawable build =
+                        new DrawableCreator.Builder()
+                                .setCornersRadius(AutoSizeUtils.dp2px(mContext, 20F))
+                                .setSolidColor(backgroundColor)
+                                .build();
+                view.setBackground(build);
```
Comment (QuincySx) [inline android/app/src/main/java/org/haobtc/onekey/adapter/WalletListTypeAdapter.java]: 使用动态 背景，和在 AssetsLogo 里获取颜色

File: android/app/src/main/java/org/haobtc/onekey/adapter/WalletListAdapter.java [hunk 6]
```diff
@@ -35,19 +34,28 @@ public WalletListAdapter(@Nullable List<LocalWalletInfo> data) {
...
+                backgroundColor = mContext.getColor(R.color.color_f0b90b);
+                break;
+            case HECO:
+                backgroundColor = mContext.getColor(R.color.color_01943f);
                 break;
             default:
             case BTC:
-                view.setBackground(mContext.getDrawable(R.drawable.orange_back));
-                imgType.setImageDrawable(mContext.getDrawable(R.drawable.token_trans_btc_list));
+                backgroundColor = mContext.getColor(R.color.text_nine);
```
Comment (QuincySx) [inline android/app/src/main/java/org/haobtc/onekey/adapter/WalletListAdapter.java]: 使用 AssetsLogo 类获取颜色

File: android/app/src/main/java/org/haobtc/onekey/adapter/WalletListTypeAdapter.java [hunk 7]
```diff
@@ -50,38 +49,16 @@ protected void convert(BaseViewHolder helper, WalletInfo item) {
...
-                                    null));
                 } else if (item.mCoinType == Vm.CoinType.HECO) {
                     backgroundColor = mContext.getColor(R.color.color_01943f);
-                    imgType.setImageDrawable(
-                            ResourcesCompat.getDrawable(
-                                    helper.itemView.getResources(),
-                                    R.drawable.vector_token_heco,
-                                    null));
                 }
-
```
Comment (QuincySx) [inline android/app/src/main/java/org/haobtc/onekey/adapter/WalletListTypeAdapter.java]: 使用 AssetsLogo 类获取颜色

File: android/app/src/main/java/org/haobtc/onekey/adapter/HdWalletAssetAdapter.kt [hunk 8]
```diff
@@ -27,39 +27,35 @@ class HdWalletAssetAdapter(context: Context, data: List<BalanceCoinInfo?>?) : Ba
...
+        helper.setText(R.id.text_balance, getFormatBalance(item, 6))
       }
       else -> {
         RemoteImage(item.icon).intoTarget(helper.getView(R.id.imageView))
-        helper.setText(R.id.text_wallet_name, item.coin)
-          .setText(R.id.text_balance, getFormatBalance(item, 4))
+        helper.setText(R.id.text_balance, getFormatBalance(item, 4))
       }
     }
+    helper.setText(R.id.text_wallet_name, item.coin.toUpperCase())
```
Comment (QuincySx) [inline android/app/src/main/java/org/haobtc/onekey/adapter/HdWalletAssetAdapter.kt]: 使用 AssetsLogo 类获取颜色，CoinDisplayUtils 格式化数据展示

File: android/app/src/main/java/org/haobtc/onekey/adapter/WalletListAdapter.java (hunk 6 above)
Comment (peterQin0826) [reply android/app/src/main/java/org/haobtc/onekey/adapter/WalletListAdapter.java]: 这个在后面的提交里面

File: android/app/src/main/java/org/haobtc/onekey/onekeys/homepage/WalletFragment.java (hunk 3 above)
Comment (peterQin0826) [reply android/app/src/main/java/org/haobtc/onekey/onekeys/homepage/WalletFragment.java]: 后面的commit 已经修改了

File: android/app/src/main/java/org/haobtc/onekey/adapter/WalletListTypeAdapter.java (hunk 5 above)
Comment (peterQin0826) [reply android/app/src/main/java/org/haobtc/onekey/adapter/WalletListTypeAdapter.java]: 已修改

File: android/app/src/main/java/org/haobtc/onekey/adapter/HdWalletAssetAdapter.kt (hunk 1 above)
Comment (peterQin0826) [reply android/app/src/main/java/org/haobtc/onekey/adapter/HdWalletAssetAdapter.kt]: 已修改，新的commit 体现

**Review by QuincySx — APPROVED** [review]

**Review by liyanhrxy — APPROVED** [review]

**Pull request merged**